### PR TITLE
feat(ui): Replace releases v1 with v2

### PR DESCRIPTION
**Pull request by matejminar**
This PR makes Releases v2 default for opted in users.
Releases v1 are hidden from the sidebar for them.
They can still opt out of v2 by clicking on the "Go to Legacy Releases" button.

Urls for both versions are the same - we look at organization features and localstorage to decide which component to serve.

#sync-getsentry

### CODE REVIEW

File: src/sentry/static/sentry/app/routes.jsx [hunk 1]
```diff
@@ -1257,194 +1259,228 @@ function routes() {
...
-          path="/organizations/:orgId/releases-v2/"
+          path="/organizations/:orgId/releases/"
           componentPromise={() =>
-            import(/* webpackChunkName: "ReleasesV2Container" */ 'app/views/releasesV2')
+            decideReleasesVersion(v2 =>
+              v2
+                ? import(
+                    /* webpackChunkName: "ReleasesV2Container" */ 'app/views/releasesV2'
+                  )
+                : import(/* webpackChunkName: "ReleasesContainer" */ 'app/views/releases')
```
Comment (markstory) [inline src/sentry/static/sentry/app/routes.jsx:1345]: A nice benefit of this approach is that removing releases v1 should be safe and simple.

File: src/sentry/static/sentry/app/views/releases/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  orgId: string;
+};
+
+type Props = RouteComponentProps<RouteParams, {}>;
+
+class ReleasesContainer extends React.Component<Props> {
+  render() {
+    const {children} = this.props;
+
+    return children;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/releases/index.tsx:14]: Why do you need this component if it doesn't really do anything. It is so you have a place to attach nested routes to?

File: src/sentry/static/sentry/app/views/releasesV2/detail/index.tsx [hunk 3]
```diff
@@ -217,5 +219,11 @@ const ProjectsFooterMessage = styled('div')`
   grid-gap: ${space(1)};
 `;
 
+const ContentBox = styled('div')`
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/releasesV2/detail/index.tsx:222]: Nice job pulling this up :clap:

File: src/sentry/static/sentry/app/views/releases/index.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/releases/index.tsx:14]: Yes, I wanted to have the exact same structure in v1 and v2

**Comment by markstory** [comment]
Percy diffs look relevant, you might need to update/remove some acceptance tests.

**Comment by matejminar** [comment]
Yes, there was still 'v2' in the acceptance test URL path - it should be fine now.

**Review by markstory — APPROVED** [review]
Looks good to me. Matej and I did a walkthrough yesterday as well.

**Pull request merged**